functions/go/fix/fixpkg/fix.go

```diff
@@ -0,0 +1,453 @@
+package fixpkg
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-openapi/spec"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/fieldmeta"
+	"sigs.k8s.io/kustomize/kyaml/fn/runtime/runtimeutil"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+	"sigs.k8s.io/kustomize/kyaml/sets"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/fix/v1alpha1"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/fix/v1alpha2"
+)
+
+var _ kio.Filter = &Fix{}
+
+// Fix migrates the resources in v1alpha1 format to v1alpha2 format
+type Fix struct {
+	// settersSchema is the schema equivalent of openAPI section in Kptfile
+	settersSchema *spec.Schema
+}
+
+// Filter implements Fix as a yaml.Filter
+func (s *Fix) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	kfFound := false
+	// the input nodes are sorted based on package path depth with Kptfile first in each package
```

Comment:
So this seems to be based on some assumption around the order in which resources will be provided. Can we make that explicit in the comment of the function?

Reply:
Good catch. I refactored the function to not depend on input resources order. The resources are grouped according to packages.